### PR TITLE
feat: adds badge base style

### PR DESCRIPTION
**Pull request by kendallstrautman**
## 🔗 Relevant links

- [Preview link](https://dev-portal-git-ksadds-base-badge-style-hashicorp.vercel.app/swingset/components/badge) 🔎
- [Asana task](https://app.asana.com/0/home/1199634971443477/1204604036286170) 🎟️

## 🗒️ What

Adds a new badge base style that mimics the outline style but without the container. This is needed for the updated badges in the tutorial metadata with variants work. This isn't a sanctioned style by HDS but one that could be proposed to be added.

This approach was preferred over copying over all badge behavior to the unique metadata usecase since it seems like a useful variation that could be used in the future.  

## 🤷 Why

Adds a 'base' type option to the `Badge` component, mimics the outline type classes but without the border styles. 

Screenshot 2023-05-16 at 9 55 29 AM

## 📸 Design Screenshots

See usecase example in the [tutorial variants metadata work](https://www.figma.com/file/NAFjIPkbwjyY8jPl6T9nlP/Tutorial-Variants?type=design&node-id=172-31955&t=PlF0QTBCVhYIMs2w-0), there isn't a full spec for this style yet. 
Screenshot 2023-05-16 at 9 55 07 AM

## 🧪 Testing

- [Visit swingset, ](https://dev-portal-git-ksadds-base-badge-style-hashicorp.vercel.app/swingset/components/badge) validate that the new 'base' type styles are there and valid. 

## 💭 Anything else?

This PR doesn't use the new style, just adds the functionality to support it. The use of this new style will come in a follow up PR with tutorial metadata rework.

### CODE REVIEW

File: src/components/badge/badge.module.css [hunk 1]
```diff
@@ -73,6 +73,10 @@
 ***
 */
 
+.base {
+	padding: 0;
```
Comment (kendallstrautman) [inline src/components/badge/badge.module.css:77]: Since there isn't a container or outline with this style, the padding isn't needed.

**Review by zchsh — APPROVED** [review]
✅ New style looks good to me! I checked out [swingset](https://dev-portal-git-ksadds-base-badge-style-hashicorp.vercel.app/swingset/components/badge), I dig that "sticker sheet" table of variants 🙌 

---

💭 Very optional query, maybe more design than code related - is this an additional variant we expect to appear at some point "upstream" in [HDS Badge](https://helios.hashicorp.design/components/badge)? Or is this more of an intentional dev-dot-specific divergence from HDS?

**Comment by kendallstrautman** [comment]
> is this an additional variant we expect to appear at some point "upstream" in [HDS Badge](https://helios.hashicorp.design/components/badge)? Or is this more of an intentional dev-dot-specific divergence from HDS?

Tagging @braydenlove here to give more input but I believe Brayden is going to show it to Jory to get the convo going with HDS. But at this point it is an intentional deviation for dev dot purposes I believe. I think this points to a lack of clarity on process of how we should propose new variations to HDS.

**Comment by braydenlove** [comment]
Yup I will show Jory what we are doing here and will report back if we need to update anything 👍🏼

**Review by braydenlove — APPROVED** [review]

**Pull request merged**